.github/workflows/dev-deploy.yaml

```
@@ -0,0 +1,249 @@
+                  git diff --name-only HEAD^ HEAD > changed_files.txt
+                  echo "Changed files:"
+                  cat changed_files.txt
+
+                  if grep -q -E '^(fitnessllm_dataplatform/|Dockerfile|pyproject.toml|poetry.lock)' changed_files.txt; then
+                    echo "docker_build_needed=true" >> $GITHUB_OUTPUT
+                  else
+                    echo "docker_build_needed=false" >> $GITHUB_OUTPUT
+                  fi
+
+            - name: Set up Python
+              uses: actions/setup-python@v4
```

actionlint (1.7.4)

the runner of "actions/setup-python@v4" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)